src/graphdb.js

```diff
@@ -52,74 +52,157 @@ function toPascalCase (str) {
+        let nodeCase = replaceCleanseLabel(node.label);
+
+        if (changeCase && (node.label !== nodeCase) || changeCase) {
+            r += `type ${toPascalCase(nodeCase)} @alias(property:"${node.label}") {\n`;
+        }
+        else if (node.label !== nodeCase) {
             r += `type ${nodeCase} @alias(property:"${node.label}") {\n`;
-        } else {
+        }
+        else {
             r += `type ${nodeCase} {\n`;
         }
         
         r += '\t_id: ID! @id\n';
 
         node.properties.forEach(property => {
+            let propertyCase = replaceCleanseLabel(property.name);
             if (property.name == 'id')
                 r+= `\tid: ID\n`;
+            else if (property.name !== propertyCase) {
+                r+= `\t${propertyCase}: ${property.type} @alias(property: "${property.name}")\n`;
+            }
             else
                 r+= `\t${property.name}: ${property.type}\n`;
         });
         
         let edgeTypes = [];
         gdbs.edgeStructures.forEach(edge => {            
             edge.directions.forEach(direction => {
-                let fromCase = toPascalCase(direction.from);
-                let toCase = toPascalCase(direction.to);
-                let edgeCase = toPascalCase(edge.label);
+                nodeCase = replaceCleanseLabel(node.label);
+                invalidNode = node.label !== nodeCase ? true : false;
+                nodeCase = toPascalCase(nodeCase);
+
+                let fromCase = replaceCleanseLabel(direction.from);
+                let invalidDirFrom = direction.from !== fromCase ? true : false;
+                fromCase = toPascalCase(fromCase);
+
+                let toCase = replaceCleanseLabel(direction.to);
+                let invalidDirTo = direction.to !== toCase ? true : false;
+                toCase = toPascalCase(toCase);
+
+                let edgeCase = replaceCleanseLabel(edge.label);
+                let invalidEdge = edge.label !== edgeCase ? true : false;
+                edgeCase = toPascalCase(edgeCase);
+
                 if (direction.from == node.label && direction.to == node.label){
                     if (direction.relationship == 'MANY-MANY') {
-                        r += `\t${nodeCase.toLocaleLowerCase() + edgeCase}sOut(filter: ${nodeCase}Input, options: Options): [${nodeCase}] @relationship(edgeType:"${edge.label}", direction:OUT)\n`;
-                        r += `\t${nodeCase.toLocaleLowerCase() + edgeCase}sIn(filter: ${nodeCase}Input, options: Options): [${nodeCase}] @relationship(edgeType:"${edge.label}", direction:IN)\n`;                    
+                        if (invalidNode || invalidEdge) {
+                            r += `\t${nodeCase.toLocaleLowerCase() + edgeCase}sOut(filter: ${nodeCase}Input, options: Options): [${nodeCase}] @relationship(edgeType:"${edge.label}", direction:OUT) @alias(property: "${node.label.toLocaleLowerCase() + toPascalCase(edge.label)}sOut")\n`;
```

Comment:
I don't think this code produces a proper query. When I try to select an edge relationship to retrieve in the app sync UI it always produces null results
![Screenshot 2024-11-06 at 10 30 46 AM](https://github.com/user-attachments/assets/2a65bc07-bf68-4b47-bfba-f00386a4eeae)

Reply:
On my local environment it worked if I back ticked every reference to an alias, `schemaTypeInfo.typeAlias`, `egdgeTypeAlias` (the latter is probably a typo).